### DISCUSSION TITLE
v6网络有关问题

### DISCUSSION THREAD

**Discussion by LYOfficial** [CakeAL/ustb-wifi-tools#6] [General]
在你科注销页显示的V6流量标题为 **登录前流量(V6):xxxMB** 

想知道这个v6流量是实时统计的还是只计算登录前消耗的v6流量？

还有一个问题，就是你科校园网v4和v6是一体的，如果在网络控制关闭IPV4只保留IPV6，校园网就会断线，原因大概是不能访问到注销页登录账号，有没有什么方法保持自己账号登录，在不跑v4流量的情况下放心使用IPV6？

**Comment by CakeAL** [comment]
1. **登录前流量(V6):xxxMB**（参考我的应用里面`月度使用概览`里面说的），因为校园网每次都需要等你下线之后，才能记录一次数据，之后把当月的累加，就是所谓的**登录前**，也就是说注销页的数据不是实时的，就是距你这次联上校园网之前，ipv4和ipv6的下行（这俩数据没有任何参考意义，而且注销页变成登录页登录还会R什么玩应超时，那个是因为那个页面获取不到登录服务器ip的问题，反正气死，所以才做了这个应用）。
2. 首先，原因不是不能访问注销页，校园网判断你是否登录是看在校园网登记的设备的MAC地址，而不是每次你连上校园网，就会从登录页/注销页登录一次。登录校园网之后，关闭ipv4，校园网不会断开，只用ipv6是可以的，比如访问纯ipv6网站(https://mirrors6.tuna.tsinghua.edu.cn/ )，但是你需要设置ipv6 DNS来提供ipv6的域名解析服务，因为默认情况下没有设置的话，需要先向ipv4的DNS服务器发查询ip的请求，但是此时ipv4关闭，所以无法获取域名对应的地址。但是有ipv6 DNS的话就可以直接对域名解析。或者使用直接ipv6访问例如(https://[2001:da8:208:10::14]/ )。关于不跑v4流量情况下使用v6，这是不可能的，因为现在整个互联网底层还是ipv4，比如qq这种应用，他是直接通过ip发送的信息，这些ip都是ipv4，你无论如何都不可能在现有情况下把他变成ipv6（除非把这些ipv4流量代理走，如果你有ipv6代理服务器的话）

**Reply by LYOfficial** [reply]
好的，谢谢喵 
![a49f9a3c3c70549cba4dd5074b804d97](https://github.com/user-attachments/assets/b77acc83-0108-4bdc-b9ee-670be04391a9)

**Comment by CakeAL** [comment]
你可以试一下连接校园网，然后断开ipv4，上述两个网站仍可以访问。第一个需要配置ipv6 DNS，第二个不需要配置ipv6 DNS

**Comment by firezhuo-chen** [comment]
你科的校园网做了AP隔离，PC和移动设备即使同处校园网也无法通过IPv6进行远程桌面和串流，是否有解决办法？

还有就是校园网貌似只给设备发一个128位前缀的IPv6地址，无法继续划分子网，在宿舍使用Openwrt路由器的场景下，有无方法使接入的设备正常使用IPv6？（或者拿到一个/64的IPv6地址？）

**Reply by CakeAL** [reply]
使用tailscale或者zerotier建立隧道，但是貌似不能只走ipv6，会自动坍缩成ipv4隧道（然后导致这个流量去公网绕了一圈，回来还是算你校园网下行）。
不知道。
